src/__tests__/auto-cleanup.js

```diff
@@ -25,9 +25,22 @@ class Test extends React.Component<*> {
 test('component is mounted, but not umounted before test ends', () => {
   const fn = jest.fn();
   render(<Test onUnmount={fn} />);
+  expect(isMounted).toEqual(true);
   expect(fn).not.toHaveBeenCalled();
 });
 
 test('component is automatically umounted after first test ends', () => {
   expect(isMounted).toEqual(false);
 });
+
+test('does not time out with fake timers', () => {
+  jest.useFakeTimers();
```

Comment:
As I understand there are 3 possible values for `useFakeTimers`: `'modern'`, `'legacy'` & default. With the default changing between Jest versions. Maybe we could have 3 tests here, default, legacy & modern, WDYT?